counter-and-todo-on-redux/src/store/reducers/todo.js

```diff
@@ -0,0 +1,41 @@
+let lastId = 0;
+
+const initialState = {
+  todos: []
+}
+
+function toDoReducer (state = initialState, action) {
+  switch (action.type) {
+    case 'task/addTask':
+      return  {
+        ...state, 
+        todos: [...state.todos,{
+        id: ++lastId,
+        title: action.payload.title,
+        completed: false,
+      }]
+    }
+
+    case 'task/toggleTask':
+      return {
+        ...state,
+        todos: state.todos.map(task => {
+          if (task.id === action.payload.id) {
+            return {...task, completed: !task.completed} 
+          }
+          return task;
```

Comment:
for the cleaan code there is a common way  here


at first you write condition for exit and then do the logic

```suggestion
          if (task.id !== action.payload.id) {
             return task;
           }
 
            return {...task, completed: !task.completed} 
          }
         
```